llvm/test/CodeGen/RISCV/rvv/fixed-vectors-vaaddu.ll

```diff
@@ -39,8 +39,10 @@ define <8 x i8> @vaaddu_vv_v8i8_floor_sexti16(<8 x i8> %x, <8 x i8> %y) {
 ; CHECK-LABEL: vaaddu_vv_v8i8_floor_sexti16:
 ; CHECK:       # %bb.0:
 ; CHECK-NEXT:    vsetivli zero, 8, e8, mf2, ta, ma
-; CHECK-NEXT:    vwadd.vv v10, v8, v9
-; CHECK-NEXT:    vnsrl.wi v8, v10, 1
+; CHECK-NEXT:    vand.vv v10, v8, v9
+; CHECK-NEXT:    vxor.vv v8, v8, v9
+; CHECK-NEXT:    vsra.vi v8, v8, 1
+; CHECK-NEXT:    vadd.vv v8, v10, v8
```

Comment:
@topperc Would it be acceptable to just add avgfloors/avgflooru custom lowering to riscv? I'm not sure how better we can make use the vwadd widening add instruction.

Reply:
@topperc @preames These are the last outstanding regressions in the draft patch before I start cleanup - do you have any suggestions on the preferred way that riscv should handle AVGFLOORS? AVGCEILS appears to be fine.

Reply:
We can add custom lowering. Ideally we'd use vaadd, but that requires a write to a rounding mode register that some microarchitectures haven't optimized.

Reply:
Oh I guess we do use vaaddu for some cases already. I guess we just need to do the rest.

Reply:
Does AVGFLOORS of -1 and 0 return -1? I think that's what stopped this from getting implemented before. There was maybe confusion around the meaning of "average". https://github.com/riscv/riscv-v-spec/issues/935

Reply:
Yes, it should return -1:
```
avgfloors(lhs, rhs) -> add(and(lhs,rhs),ashr(xor(lhs,rhs),1))
avgfloors(lhs, rhs) -> trunc(ashr(add(sext(lhs),sext(rhs)),1))
avgfloors(lhs, rhs) -> trunc(lshr(add(sext(lhs),sext(rhs)),1))
```

Reply:
Thanks @topperc